network/src/protocols/discovery/mod.rs

```diff
@@ -512,7 +527,7 @@ fn is_valid(
     )?;
 
     let peer_info = PeerInfo::decode(peer_info_bytes)?;
-    for addr in peer_info.addrs {
+    for addr in &peer_info.addrs {
         let _: Multiaddr = Multiaddr::try_from(addr.clone())?;
```

Comment:
Can you also create a VerifiedPeerInfo since we are validating here that the addr in the peer info is correctly parsed as a `Multiaddr`?